Check for Video Capture Device interference on Windows

prgmitchell opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->
This checks to see if a user has added multiple video capture devices pointing to the same device on Windows, then throws a warning if they do.

![firefox_Mh6v67hzlU](https://user-images.githubusercontent.com/86465454/221981543-d6f9a737-b115-487b-bab4-507aea6a542e.png)
<img width="1117" alt="Screenshot 2023-03-02 at 7 11 39 AM" src="https://user-images.githubusercontent.com/86465454/222425356-d38828b1-20b3-4b57-a3d2-c84602b6c547.png">

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open GitHub Issue, or implements feature request -->
<!--- from the Ideas page, please link to the issue here. -->
This comes up often enough in support and there's no clear indication to the user that they shouldn't be doing this.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->
Tested locally with this log with these two logs:
https://obsproject.com/logs/2DVdXkIVmhXGxVeR
https://obsproject.com/logs/mi4bNDc0I3xjjf4B

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
- New feature (non-breaking change which adds functionality)
<!--- - Tweak (non-breaking change to improve existing functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
<!--- - Code cleanup (non-breaking change which makes code smaller or more readable) -->
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
<!--- - Documentation (a change to documentation pages) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

RytoEX commented on checks/windows.py:
```diff
@@ -350,3 +350,37 @@ def checkWindowsARM64EmulationStatus(lines):
+
+        if (duplicateList):
+            return [LEVEL_WARNING, "Video Capture Device Interference",
+                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding sources to new scenes."]
```
Is this language pulled from the support bot or is it brand new?

prgmitchell replied:
It is new, currently we have no checks or bot commands related to this.

prgmitchell commented:
I have decided to update this PR to include a list of sources that would make it more clear to the user where the issue is.

RytoEX commented on checks/windows.py:
```diff
@@ -350,3 +351,50 @@ def checkWindowsARM64EmulationStatus(lines):
+
+        if (duplicateList):
+            return [LEVEL_WARNING, "Video Capture Device Interference",
+                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding sources to new scenes.<br><br>The following source list contains duplicates:<br>" + dupListString]
```
```suggestion
                    "You have multiple Video Capture Device sources pointed to the same device. This is known to cause issues such as Video Capture Device sources working intermittently or not working at all. Remove any duplicates and use the <strong>Add Existing</strong> option when adding the same Video Capture Device to new scenes.<br><br>The following source list contains duplicates:<br>" + dupListString]
```

prgmitchell replied:
PR updated with this wording.

prgmitchell commented:
I have updated the PR for the wording changes requested. I have also updated the code to group duplicate devices if the user has 2 sources pointed at one device and 2 sources pointed at another for example. This makes it easier for a user to understand which sources are pointed at which device if they have multiple. I have used an <hr> border for this and matched the color with the log analyzer CSS.

prgmitchell commented:
I have converted this to a draft as I found a bug in the original code.

prgmitchell commented:
Having had some time to think about this, I am going to close this PR. Because a user can change their capture device settings and they are logged each time, most users logs will result in this flag if they are changing capture device settings trying to get it working.

RytoEX commented:
Fair enough. I'd meant to follow-up and ask about how it handles multiple device settings changes, but this got lost in my to-do list.
